subprojects/core/src/main/java/org/gradle/api/tasks/SourceTask.java

```diff
@@ -68,7 +69,7 @@ protected PatternFilterable getPatternSet() {
     @InputFiles
     @SkipWhenEmpty
     @IgnoreEmptyDirectories
-    @ToBeReplacedByLazyProperty
+    @NotToBeReplacedByLazyProperty(because = "SourceTask will be deprecated")
     @PathSensitive(PathSensitivity.ABSOLUTE)
     public FileTree getSource() {
```

Comment:
💭 The (implicit) contract for `SourceTask` is that implementors should override `getSource()`. That directly contradicts our new validation.